finality-aleph/src/data_io.rs

```diff
@@ -377,8 +377,10 @@ pub(crate) async fn refresh_best_chain<B, BE, SC, C>(
     // uses is that once the block in `proposed_block` reaches the height of `max_block_num`, and the just queried
     // `best_block` is a `descendant` of the previous query, then we don't need to update proposed_block, as it is
     // already correct.
-    let mut prev_best_hash: B::Hash = proposed_block.lock().hash;
-    let mut prev_best_number: NumberFor<B> = proposed_block.lock().number;
+    let block = proposed_block.lock().await;
+    let mut prev_best_hash: B::Hash = block.hash;
+    let mut prev_best_number: NumberFor<B> = block.number;
+    drop(block);
```

Comment:
Rather than explicitly `drop`ping it might be better to enclose this into a section – I think something like this would work:
```
let (prev_best_hash, prev_best_numbere) = async {
 let block = proposed_block.lock().await;
 (block.hash, block.number)
}.await;
```
Or perhaps just `.clone()` the whole thing and extract things from that.

BIg alternative solution – keep data about the previous block separately from the mutex and use it for updating, just modifying the mutex by copying the current state into it as the last step of the whole computation process.

Reply:
I added an async block (I think it looks beautiful :3 ). The alternative solution is not in scope of this PR and I guess @DamianStraszak is working on it right now?